cas/src/main/java/org/springframework/security/cas/web/CasAuthenticationEntryPoint.java

```diff
@@ -98,7 +98,7 @@ protected String createServiceUrl(HttpServletRequest request, HttpServletRespons
 	 */
 	protected String createRedirectUrl(String serviceUrl) {
 		return CommonUtils.constructRedirectUrl(this.loginUrl, this.serviceProperties.getServiceParameter(), serviceUrl,
-				this.serviceProperties.isSendRenew(), false);
+				this.serviceProperties.isSendRenew(), this.serviceProperties.isGateway());
```

Comment:
Is it true that we should always send `gateway=true` to CAS, even if this is not a gateway request? In other words, if I do `serviceProperties.setGateway(true)` does that mean that every redirect to CAS will include `gateway=true`?

Reply:
Yes, it means that all calls to CAS using **these** entry point and `serviceProperties` will have `gateway=true`, but you can have two `serviceProperties` (with and without `gateway`) and two entry points to have URLs in your client app where you use `gateway` and other URLs where you don't.

The use case would a public/protected home page (with `gateway`) and a protected area for the rest of the webapp (without `gateway`).